charbot/levels.py

```diff
@@ -149,13 +149,13 @@ async def proc_xp(self, message: discord.Message):
                 if gained + user["detailed_xp"][0] >= self._xp_function(user["level"]):
                     new_level = user["level"] + 1
                     detailed = [0, self._xp_function(new_level), user["xp"] + gained]
-                    new_xp = detailed[2]
+                    new_xp = detailed[0]
                     await message.channel.send(
                         f"{message.author.mention} has done some time, and is now level **{new_level}**."
                     )
                     await update_level_roles(member, new_level)
                 else:
-                    detailed = user["detailed_xp"]
+                    detailed[0] = user["detailed_xp"]
```

Comment:
There is a different way to do this, via unpacking and then repacking, if you want to do it on one line.
```suggestion
                    detailed = user["detailed_xp"]
                    detailed[0] += gained
                    detailed[2] += gained
```